server/hotrod/src/main/java/org/infinispan/server/hotrod/CacheRequestProcessor.java

```diff
@@ -542,6 +543,7 @@ void iterationStart(HotRodHeader header, Subject subject, byte[] segmentMask, St
          try {
             String iterationId = server.getIterationManager().start(cache, segmentMask != null ? BitSet.valueOf(segmentMask) : null,
                   filterConverterFactory, filterConverterParams, header.getValueMediaType(), batch, includeMetadata);
+            channel.closeFuture().addListener(new IterationReaper(server.getIterationManager(), cache.getName(), iterationId));
```

Comment:
This is a memory leak as it will allocate an `IterationReaper` per iteration. If this client lives for a long time this can keep any number of those objects alive until it finally disconnects.

Reply:
Good eyes there sir !